### DISCUSSION TITLE
Android apk poor performance

### DISCUSSION THREAD

**Discussion by s-moraes** [ggml-org/llama.cpp#7017] [General]
I am trying to embedded llama.cpp inside my apk, but for some reason it is very slow.

But when using a rooted device (pixel 6), to building and execute directly from adb shell (native binary, not apk), I am getting a pretty nice performance... So I am wordering if there is any difference running from native binary from shell and apk using the library.

Is it possible to have good performance for apps not running via adb shell?

**Comment by gardner** [comment]
It would be easier to tell what's going on by inspecting the build logs.

Is it possible one build is using vulkan or some other optimization?
